corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js

```diff
@@ -459,5 +460,67 @@ hqDefine("cloudcare/js/formplayer/utils/utils", function () {
         });
     };
 
+    Utils.setSyncInterval = function (appId, restartInterval) {
+        hqRequire(["cloudcare/js/formplayer/app"], function (FormplayerFrontend) {
+            const currentApp = FormplayerFrontend.getChannel().request("appselect:getApp", appId);
+            let customProperties = {};
+            if (currentApp && currentApp.attributes && currentApp.attributes.profile) {
+                customProperties = currentApp.attributes.profile.custom_properties || {};
+            }
+
+            const useAggressiveSyncTiming = (customProperties[constants.POST_FORM_SYNC] === "yes");
+            if (!useAggressiveSyncTiming) {
+                return;
+            }
+
+            const FIVE_MINUTES_IN_MILLISECONDS = 1000 * 60 * 5;
+            if (restartInterval) {
+                stopSyncInterval();
+                startSyncInterval(FIVE_MINUTES_IN_MILLISECONDS);
+            } else {
+                startSyncInterval(FIVE_MINUTES_IN_MILLISECONDS);
+            }
+        });
+    };
+
+    function startSyncInterval(delayInMilliseconds) {
+        function shouldSync() {
+            const currentTime = Date.now(),
+                lastUserActivityTime =  sessionStorage.getItem("lastUserActivityTime") || 0,
+                elapsedTimeSinceLastActivity = currentTime - lastUserActivityTime,
+                isInApp = Utils.currentUrlToObject().appId !== undefined;
+            if (elapsedTimeSinceLastActivity <= delayInMilliseconds && isInApp) {
```

Comment:
should this be `elapsedTimeSinceLastActivity >= delayInMilliseconds`, like if more time has elapsed since last activity than the delay time, then we should sync?

Reply:
I can see how that's confusing - i thought the same thing as you looking back at it again. But I think this is correct. The logic here is that we should only sync if the user has been active within the last 5 minutes (`delayInMilliseconds`). If the user has not been active in the last 5 minutes, then we don't want to sync. This to prevent continuously syncing if a user has web apps open but it's in the background and not actively used.

Reply:
ahh, ok. Thanks for clarifying!